test/integration_test/basic_dmthin_test.go

```diff
@@ -22,124 +22,259 @@ import (
 )
 
 var (
-	pxVer2_13_0, _ = version.NewVersion("2.13.0")
+	pxVer3_0_0, _ = version.NewVersion("3.0.0")
+)
+
+type TestOptionsPresent int
+
+const (
+	PxStoreV2Present TestOptionsPresent = iota
+	MetadataPresent
+	AutoJournalPresent
+	KvdbDevicePresent
+	JournalPresent
+	MaxOptions
 )
 
-func defaultDmthinSpec(t *testing.T) *corev1.StorageCluster {
+type OptionsArr [MaxOptions]bool
+
+func defaultDmthinSpec(t *testing.T, addDmthinOption bool) *corev1.StorageCluster {
 	objects, err := ci_utils.ParseSpecs("storagecluster/storagecluster-with-all-components.yaml")
 	require.NoError(t, err)
 	cluster, ok := objects[0].(*corev1.StorageCluster)
 	require.True(t, ok)
 	cluster.Name = "test-stc"
 	cluster.Namespace = "kube-system"
 	cluster.Spec.Monitoring.Prometheus.AlertManager.Enabled = false
-	cluster.Annotations = map[string]string{
-		"portworx.io/misc-args": "-T PX-StoreV2",
+	if addDmthinOption {
+		cluster.Annotations = map[string]string{
+			"portworx.io/misc-args": "-T PX-StoreV2",
+		}
 	}
 	return cluster
 }
 
-var testDmthinCases = []types.TestCase{
-	{
-		/* Install DMthin by specifying only metadata device */
-		TestName:        "BasicInstallDmthinWithMetadataDevice",
-		TestrailCaseIDs: []string{""},
-		TestSpec: func(t *testing.T) interface{} {
-			provider := cloud_provider.GetCloudProvider()
-			cluster := defaultDmthinSpec(t)
-			cloudSpec := &corev1.CloudStorageSpec{}
-			cloudSpec.DeviceSpecs = provider.GetDefaultDataDrives()
-			cloudSpec.SystemMdDeviceSpec = provider.GetDefaultMetadataDrive()
-			cluster.Spec.CloudStorage = cloudSpec
-			return cluster
-		},
-		TestFunc: BasicInstallDmthin,
-	},
-	{
-		/* Install DMthin by specifying metadata and kvdb devices */
-		TestName:        "BasicInstallDmthinWithMetadataAndKVDBDevice",
-		TestrailCaseIDs: []string{""},
-		TestSpec: func(t *testing.T) interface{} {
-			provider := cloud_provider.GetCloudProvider()
-			cluster := defaultDmthinSpec(t)
-			cloudSpec := &corev1.CloudStorageSpec{}
-			cloudSpec.DeviceSpecs = provider.GetDefaultDataDrives()
+func getTestName(opt OptionsArr) string {
+	testName := "BasicInstallDmthinWith"
```

Comment:
[nit] Looks like this method can sometimes result in a test just named `TestStorageClusterDmthinWithoutPxStoreV2Option/BasicInstallDmthinWith`: can we add a final `else` to make it return `WithNothing` instead?